src/components/HorizontalStepper.tsx

```diff
@@ -187,37 +198,37 @@ export const HorizontalStepper: FC<HorizontalStepperProps> = (props) => {
         maxWidth="md"
         flexDirection="row"
         py={2}
-        sx={{ backgroundColod: 'background.default' }}
+        sx={{ backgroundColor: 'background.default' }}
       >
-        {!!prevStep && (
+        {!!prevStep && !!backButton && (
           <Button
             id="horizontal-stepper-back-button"
-            onClick={onBack}
-            disabled={disabled || disabledBack}
+            onClick={backButton.onClick}
+            disabled={disabled || backButton.disabled}
           >
-            {getBackLabel(prevStep)}
+            {backButton.getLabel && backButton.getLabel(prevStep)}
```

Comment:
```suggestion
            {backButton.getLabel?.(prevStep)}
```
Acá podes hacerlo así tambien